human-viz-bridge/src/HumanJointStateBridge.cpp

```diff
@@ -0,0 +1,188 @@
+}
+
+class xsensJointStatePublisherModule : public RFModule, public TypedReaderCallback<HumanState> 
+{
+    Port client_port;
+    BufferedPort<HumanState> humanStateDataPort;
+
+    Publisher<sensor_msgs_JointState> publisher;
+    Publisher<tf2_msgs_TFMessage> publisher_tf;
+    
+    //variables declaration
+    sensor_msgs_JointState joint_state;
+    tf2_msgs_TFMessage tf;
+    
+public:
+    double getPeriod()
+    {
+        // module periodicity (seconds).
+        return 100;
+    }
+    // Main function. Will be called periodically every getPeriod() seconds
+    bool updateModule()
+    {
+        return true;
+    }
+    // Configure function. 
+    bool configure(ResourceFinder &rf)
+    {
+        string moduleName = rf.check("name", Value("human-jointstate-bridge"), "Checking module name").asString();
+        string serverName = rf.check("serverName", Value("human-state-provider"), "Checking server name").asString();
+        string urdfModelFile = rf.findFile("urdf_model");
+        setName(moduleName.c_str());
+        
+        // Load model
+        iDynTree::ModelLoader modelLoader;
+        iDynTree::Model model;
+        if (!modelLoader.loadModelFromFile(urdfModelFile)) {
+            yError() << "Failed to load URDF model from file " << urdfModelFile;
+            return false;
+        }
+        
+        model = modelLoader.model();
+
+        if (!publisher.topic("/human/joint_states")) {
+            cerr<< "Failed to create publisher to /joint_states\n";
+            return false;
+        }
+        
+        if (!publisher_tf.topic("/tf")) {
+        cerr<< "Failed to create publisher to /tf\n";
+        return -1;
+        }
+        
+        tf.transforms.resize(1);
+        tf.transforms[0].header.frame_id = "ground";
+        tf.transforms[0].child_frame_id = "Pelvis";
+        
+        vector<string> joints;
+        joints.reserve(model.getNrOfJoints());
+        for (iDynTree::JointIndex jointIndex = 0; jointIndex < model.getNrOfJoints(); ++jointIndex) {
+            string jointName = model.getJointName(jointIndex);
+            joints.push_back(jointName);
+        }
+        
+        joint_state.name.resize(model.getNrOfJoints());
+        
+        for (size_t index = 0; index < joints.size(); ++index) {
+            joint_state.name[index] = joints[index];
+        }
+        
+        yInfo() << "Joints: " << model.getNrOfJoints() << joints;
+        
+        joint_state.position.resize(model.getNrOfJoints());
+        joint_state.velocity.resize(model.getNrOfJoints());
+        joint_state.effort.resize(model.getNrOfJoints());
+        
+        humanStateDataPort.useCallback(*this);
+        string stateProviderServerName = "/" + rf.find("serverName").asString() + "/state:o";
+        string stateReaderPortName = "/" + getName() + "/state:i";
+        humanStateDataPort.open(stateReaderPortName);           
+        if (!Network::connect(stateProviderServerName.c_str(),stateReaderPortName))
```

Comment:
We should probably remove the connect from the code, or at least let the user choose from configuration file